packages/nimble-components/src/table/components/row/index.ts

```diff
@@ -0,0 +1,38 @@
+import { observable } from '@microsoft/fast-element';
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import { styles } from './styles';
+import { template } from './template';
+import type { TableFieldValue, TableRecord } from '../../types';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-row': TableRow;
+    }
+}
+
+/**
+ * A styled row that is used within the nimble-table.
```

Comment:
What do you think of marking this and `TableCell` with `@internal`?

Pro:
- it might signify to someone (or just us) that this is not meant to be used outside of Nimble

Cons: 
- this [doesn't have any official meaning](https://stackoverflow.com/questions/28588591/internal-javascript-documentation-tag-inside-react-code-is-this-jsdoc-closure) other than [a somewhat sketchy TS compiler flag](https://www.typescriptlang.org/tsconfig#stripInternal), so it's likely to be overlooked
- we mostly haven't used it on classes before, with the exception of [ThemeProvider](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/theme-provider/index.ts) for some reason

I think I talked myself out of adding it but thought I'd mention the idea in case someone else wants to argue otherwise.

Reply:
It documents intent, that's pretty much how we use it in other spots which seems better than nothing.

Best we can do until we have a way to enforce it, which may come soon. The concept of [Scoped custom element registries](https://github.com/WICG/webcomponents/blob/gh-pages/proposals/Scoped-Custom-Element-Registries.md) (which is an encapsulated custom element registry per shadow root) is probably 1-2 years out. Part of the fast-element v2 changes includes adding support for that.